### PR TITLE
tools: logger: Create log_err function

### PR DESCRIPTION
**Pull request by ktrzcinx**
Error during conversion should be logged to stderr and to
output file by default. Such a approach will speed up debugging
process when some error with conversion occurred in CI.

Signed-off-by: Karol Trzcinski <karolx.trzcinski@linux.intel.com>

### CODE REVIEW

File: tools/logger/convert.c [hunk 1]
```diff
@@ -51,6 +52,29 @@ struct proc_ldc_entry {
...
+{
+	ssize_t needed_size;
+	va_list args;
+	char *buff;
+
+	va_start(args, fmt);
+	needed_size = vsnprintf(NULL, 0, fmt, args) + 1;
+	buff = malloc(needed_size);
+	if (buff) {
+		vsprintf(buff, fmt, args);
```
Comment (paulstelian97) [inline tools/logger/convert.c]: Don't you need to va_end and va_start args again?

Comment (ktrzcinx) [reply tools/logger/convert.c]: `va_copy`  perhaps will be better, but yes there should be fix, thanks

Comment (paulstelian97) [reply tools/logger/convert.c]: True that.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by xiulipan** [comment]
@ktrzcinx I found this update make logger output very ugly.

```
$ sudo sof-logger -l /etc/sof/sof-apl.ldc
Error: abi version in /etc/sof/sof-apl.ldc file does not coincide with abi version used by logger.
Error: abi version in /etc/sof/sof-apl.ldc file does not coincide with abi version used by logger.
logger ABI Version is 4:2:0
logger ABI Version is 4:2:0
ldc_file ABI Version is 3:14:0
ldc_file ABI Version is 3:14:0
```

Can we skip to ouput Error message to the normal ouput when we find it is a std stream?

**Comment by ktrzcinx** [comment]
@xiulipan please check changes introduced in https://github.com/thesofproject/sof/pull/2604

**Pull request merged**